docs/developers/omnichain/tutorials/swap.md

```diff
@@ -74,60 +74,51 @@ contract Swap is zContract {
         bytes calldata message
     ) external virtual override onlySystem {
         // highlight-start
-        uint32 targetChainID;
-        address recipient;
-        uint256 minAmountOut;
+        address targetTokenAddress;
+        address recipientAddress;
 
         if (context.chainID == BITCOIN) {
-            targetChainID = BytesHelperLib.bytesToUint32(message, 0);
-            recipient = BytesHelperLib.bytesToAddress(message, 4);
+            targetTokenAddress = BytesHelperLib.bytesToAddress(message, 0);
+            recipientAddress = BytesHelperLib.bytesToAddress(message, 20);
         } else {
-            (
-                uint32 targetChainID_,
-                address recipient_,
-                uint256 minAmountOut_
-            ) = abi.decode(message, (uint32, address, uint256));
-            targetChainID = targetChainID_;
-            recipient = recipient_;
-            minAmountOut = minAmountOut_;
+            (address targetToken, address recipient) = abi.decode(
+                message,
+                (address, address)
+            );
+            targetTokenAddress = targetToken;
+            recipientAddress = recipient;
         }
 
-        address targetZRC20 = systemContract.gasCoinZRC20ByChainId(
-            targetChainID
-        );
-
         uint256 outputAmount = SwapHelperLib._doSwap(
             systemContract.wZetaContractAddress(),
             systemContract.uniswapv2FactoryAddress(),
             systemContract.uniswapv2Router02Address(),
             zrc20,
             amount,
-            targetZRC20,
-            minAmountOut
+            targetTokenAddress,
+            0
```

Comment:
why did you remove the minOut?

Reply:
We weren't using it in the tutorial (it was just always 0) as it's not crucial to the core functionality.